crates/miden-testing/src/tx_context/builder.rs

```diff
@@ -243,13 +245,13 @@ impl TransactionContextBuilder {
     /// If no transaction inputs were provided manually, an ad-hoc MockChain is created in order
     /// to generate valid block data for the required notes.
     pub fn build(self) -> anyhow::Result<TransactionContext> {
-        // TODO: SourceManager.
-        let source_manager =
-            alloc::sync::Arc::new(miden_objects::assembly::DefaultSourceManager::default())
-                as alloc::sync::Arc<
-                    dyn miden_objects::assembly::SourceManager + Send + Sync + 'static,
-                >;
+        self.build_with_source_manager(default_source_manager_arc_dyn())
+    }
 
+    pub fn build_with_source_manager(
+        self,
+        source_manager: Arc<dyn SourceManagerSync>,
+    ) -> anyhow::Result<TransactionContext> {
```

Comment:
Does this need to be a _build_ method or can we make it a bit more convenient and also have a builder-style method `TransactionContext::source_manager(source_manager: Arc<dyn SourceManager>) -> Self`?

It would also be nice if we could use this in a few tests to make sure it works (which has to be checked manually afaict).